### PR TITLE
feat(workspace): priority load opened page

### PR DESCRIPTION
**Pull request by EYHN**

### CODE REVIEW

**Comment by EYHN** [comment]
Current dependencies on/for this PR:
* canary
  * **PR #5156** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5156?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5156?utm_source=stack-comment).

File: packages/frontend/workspace/src/providers/sync/peer.ts [hunk 1]
```diff
@@ -70,7 +71,8 @@ export class SyncPeer {
 
   constructor(
     private readonly rootDoc: Doc,
-    private readonly storage: Storage
+    private readonly storage: Storage,
+    private readonly priorityTarget = new PriorityTarget()
```
Comment (forehalo) [inline packages/frontend/workspace/src/providers/sync/peer.ts]: better put it in queue

Comment (pengx17) [reply packages/frontend/workspace/src/providers/sync/peer.ts]: agreed. maybe it will be easier to dedup the orchestration logic in line 228 ~ 241 etc

Comment (EYHN) [reply packages/frontend/workspace/src/providers/sync/peer.ts]: I created a PriorityAsyncQueue to implement these logic.

**Review by pengx17 — APPROVED** [review]

**Review by joooye34 — APPROVED** [review]

**Pull request merged**